### DISCUSSION TITLE
Maintain WPA/TKIP

### DISCUSSION THREAD

**Discussion by T-E-N** [Freetz-NG/freetz-ng#613] [Allgemein]
WPA/TKIP has been broken, removed since FRITZ!OS 7.21 - and missed by many who need to keep certain legacy clients connected no matter the level of security: https://thinkpad-forum.de/threads/vorsicht-mit-fritz-os-7-21-bei-nutzung-alter-wlan-karten.227294/
Could Freetz-NG continue supporting WPA/TKIP ?

**Comment by fda77** [comment]
Nobody did it, so you have to do it. Search the needed files and replace them in the new FOS version.
But beware, AVM does not publish security related news! So maybe they fixed silently a bug and you revert the fix again.

So i recommend NOT to use Fritzos at al! When your device is compatible with Openwrt you could use WPA 1-3 as you like, even with very old 7320
